cpp/src/arrow/flight/types.h

```diff
@@ -73,7 +73,7 @@ namespace flight {
 /// > all minutes are 60 seconds long, i.e. leap seconds are "smeared"
 /// > so that no leap second table is needed for interpretation. Range
 /// > is from 0001-01-01T00:00:00Z to 9999-12-31T23:59:59.999999999Z.
-using Timestamp = std::chrono::system_clock::time_point;
+using Timestamp = std::chrono::time_point<std::chrono::system_clock, std::chrono::nanoseconds>;
```

Comment:
Hmm, this is a breaking change, though - can we keep the fix on the bindings side?